src/main/kotlin/com/exactpro/th2/rptdataprovider/handlers/SearchEventsHandler.kt

```diff
@@ -310,7 +319,20 @@ class SearchEventsHandler(
             }
                 .map { it.await() }
                 .flatMapConcat { it.asFlow() }
-                .filter { it.id.toString() != request.resumeFromId }
+                .let { eventFlow: Flow<BaseEventEntity> ->
+                    when {
+                        resumeFromEvent != null -> {
+                            val dropByTimestamp = needDropByTimestamp(request, resumeFromEvent)
+                            eventFlow
+                                .dropWhile { dropByTimestamp(it) && it.id != resumeFromEvent.id }
+                                .dropWhile{ it.id == resumeFromEvent.id }
+                        }
+                        request.resumeFromId != null -> eventFlow.filter {
+                            it.id.toString() != request.resumeFromId
+                        }
```

Comment:
If I am right we never get there because the `resumeFromEvent != null` condition is true when `request.resumeFromId != null`. So the first `when` block will be always taken

Reply:
The first block depends on resumeFromId and on whether we could find this id in the database, but if the id is incorrect, the function will throw an exception, so yes, the first block is enough